src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,566 @@
+            || factors[0][0] === 0n // 0 is not prime
+            || (factors.length === 1 && factors[0][0] === -1n) // -1 not prime
+        ) {
+            return false
+        }
+        if (
+            (factors.length === 1 && factors[0][1] === 1n) // prime
+            || (factors.length === 2
+                && factors[0][0] === -1n
+                && factors[1][1] == 1n) // negative of prime
+        ) {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    //return a number which represents the color
+    primeFactors(ind: number) {
+        const factors = this.seq.getFactors(ind)
+        if (factors === null) {
+            return -30
+        } // factoring failed
+
+        //assign color to each prime number
+        const colorNum = 360 / this.countPrime
+
+        colorMap.set(1, 0)
+        let tmp = 0
+
+        for (let i = 0; i < this.primeNum.length; i++) {
+            if (colorMap.has(this.primeNum[i]) == false) {
+                tmp += colorNum
+                colorMap.set(this.primeNum[i], tmp)
+            }
+        }
+
+        //Combine color for each prime factors
+        let colorAll = -1
+        for (let i = 0; i < factors.length; i++) {
+            const thisPrime = factors[i][0]
+            const thisExp = factors[i][1]
+            for (let j = 0; j < this.primeNum.length; j++) {
+                if (thisPrime == this.primeNum[j]) {
+                    for (let k = 0; k < thisExp; k++) {
+                        if (colorAll == -1) {
+                            colorAll = colorMap.get(thisPrime)
+                        } else {
+                            colorAll =
+                                (colorAll + colorMap.get(thisPrime)) / 2
+                        }
+                    }
+                }
+            }
+        }
+
+        return colorAll
+    }
+}
+
+export const exportModule = new VisualizerExportModule(
+    'Sequence Color',
+    SeqColor,
+    ''
+)
+
+/** md
+
+## Examples
+
+Click on any image to expand it.
+
+###### The Positive Integers
+
+[<img src="../../assets/img/primes-sizes/integers.png" width="320" 
+style="margin-left: 1em; margin-right: 0.5em"
+/>](../assets/img/primes-sizes/integers.png)
+[<img src="../../assets/img/primes-sizes/ring25.png" width="320" 
+style="margin-left: 1em; margin-right: 0.5em"
+/>](../assets/img/primes-sizes/ring25.png)
+[<img src="../../assets/img/primes-sizes/ring1.png" width="320" 
+style="margin-left: 1em; margin-right: 0.5em"
+/>](../assets/img/primes-sizes/ring1.png)
+
+First, the non-negative integers, with 
+the default settings. Next, with growth rings with 
+modulus 25. Finally, with growth rings with 
+modulus 1.  The final example shows some interesting 
+effects because the rings
+actually occur more rapidly than once per pixel.  The first
+two discs appear to be missing because they have no prime factors
+to color them.
+
+###### 
```

Comment:
Still to do on documentation:  add more examples.  Note that once the grid vis is in, the latex here will work also.

Reply:
an example of semiprimes added, it's probably enough for now.

Reply:
Agreed.